src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -116,12 +110,19 @@ public FileResource(FileService fileService, ResourceLoaderService resourceLoade
      * @param keepFileName specifies if original file name should be kept
      * @return The path of the file
      * @throws URISyntaxException if response path can't be converted into URI
+     * @deprecated Implement your own usage of {@link FileService#handleSaveFile(MultipartFile, boolean, boolean)} with a mixed multipart request instead.
```

Comment:
I don’t understand the comment. Do you mean something like ‘Do not use this endpoint directly any more, instead send the file as part of a mixed multipart request to another more specific endpoint.’?

You could also link the `AttachmentUnitResource::updateAttachmentUnit` as an example of how that is done.